app/controllers/attachments_controller.rb

```diff
@@ -35,6 +35,10 @@ def create
     }
 
     if Mime::Type.lookup_by_extension(File.extname(@attachment.filename).downcase.tr('.', '')).to_s =~ /^image\//
+      image_size = ImageSize.new(uploaded_file.tempfile)
+      json[:width] = image_size.width
+      json[:height] = image_size.height
+
```

Comment:
This action is huge, while we're here can we split some of its responsibilities into helper methods or offload some of it to the `Attachment` model?